src/component/AddItem.js

```diff
@@ -3,27 +3,47 @@ import { withFirestore } from "react-firestore";
 import "../App.css";
 import Navbar from "./Navbar";
 import BackButton from "./BackButton";
+import DuplicateMessage from "./DuplicateMessage";
 
 const AddItem = ({ firestore }) => {
   const [name, setName] = useState("");
-  const [frequency, setFrequency] = useState("14");
+  const [numberOfDays, setNumberOfDays] = useState("14");
```

Comment:
Why is this a string instead of a number? I'm curious!

Reply:
We changed it back to a number on the line you have highlighted and it's working fine!

Reply:
@alejandronanez this is great question and I'm actually going to circle @mindyzwan in on this because I'm worried I may be getting my wires crossed and she wrote this last week. 

Mindy, this is set this to a string here for the radio button 'pretty soon' to be set as checked correct? Or am I'm getting mixed up? If I make this a number on line 9 the pretty soon button no longer appears to be selected when you first land on the page.

Reply:
@michellejae, setting the state there is what's telling the radio buttons which one should be checked, yes. Though actually - @ClaudiaStar implemented that part! So she's the expert and can probably explain it better than I can 😄 

My guess is, if you make it a number on line 9, you'd have to also make it a number later on in the code: e.g. `checked={numberOfDays === "14"}`. Claudia responded that you changed it to a number and it's working fine now though, yes?

Reply:
@mindyzwan my bad!! i thought she said you had written that part of the code last week -- i must have been spacing out. I'm sorry for bringing you in on this haha. 

@alejandronanez i believe I have an answer for why the string is used but want to run it by @ClaudiaStar  first as changing everything to a number does appear to cause some hiccups. Hold please. 😄 ⏱

Reply:
@michellejae and I turned the value of numberOfDays and setNumberOfDays into numbers earlier today and we thought everything was working fine BUUUT then realized that after user enters a new item, all the buttons becoming inactive. We are not entirely sure why this is happening. This is why @michellejae ended up looping @mindyzwan in on this. Maybe she knows something we don't know about how the radio buttons work...? :) Also curious to hear @alejandronanez thoughts.

Reply:
We also tried turning the value attribute to a number, i.e. (value = {7}), since when the state is set it uses the value property. This approach also makes all the buttons inactive (ie. pretty soon button does not retain the active state) after adding an item, but also more interestingly, if you take away the + before the numberOfDays const in the addItem function, a string gets saved to the database. Sorry this is all over the place, but, we are confused...haha.

Reply:
haha, oh my god, I didn't want that my comment end up opening such a big conversation, sorry about that! On the bright side, this is a great discussion between y'all 👏👏👏 

I'm a bit confused about this too, I'll check out your branch later today to see what's up!

Thanks for ALL your responses!

Reply:
@alejandronanez i'm curious to know what you find out. i was able to get it working, albeit not perfectly. 

i changed the useState to a number, and then did what @ClaudiaStar mentioned and changed any of the value attributes within the radio buttons to numbers as well, ie `value={14}`. for the checked attribute, i removed this from the _soon_ and _not soon_ buttons and changed it to true, `checked={true}` for the _pretty soon_. and lastly removed the + sign on line 44 of the current code. 

it solves the problem of saving the data to the DB as a number, and also activates the pretty soon button. however, in order to activate the _soon_ or _not soon_ buttons you have to double click it.

Reply:
The double click thing is as far as I got, too. I was even checking in the CSS for some kind of contradictory styling but I can't find anything ¯\_(ツ)_/¯

Reply:
aaaalright, so if I understand this correctly:

1. We're setting `numberOfDays` to `"14"`, which automatically checks the `Pretty Soon` checkbox, because its value is `"14"`

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L10

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L110-L117

2. Whenever we click on another checkbox, we update the state to whatever is in `event.target.value`, spoiler alert, is a string.

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L54-L56

3. We decide which checkbox to check

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L103

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L115

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L127

**Until that point, everything work as a string, and that's ok, because as you can see in point 3, we're comparing `numberOfDays` using `===` with other strings `"7" / "14" / "30"`**

4. What happens next, is when we sent the number of days, as a `Number` to Firebase, we transform the string to number by prefixing the value with `+`

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L41-L43

Mistery solved! 🙌

***

@gunderodd @mindyzwan @michellejae @ClaudiaStar thanks for all your comments, this has been a really good exercise - great communication!

Tip for the future 🔮:
- Try to stick with one data type across the whole cycle, in this case, either pick a number or a string
- You can store the _magic numbers_ `"7", "14", "30"` into constants for easier reference
- If you need to change the data type, I'd suggest you doing it here, just before setting the value using `setNumberOfDays`

https://github.com/the-collab-lab/tcl-2-smart-shopping-list/blob/a08e65709879f0f1d39164aa65c9c420e3e0f88b/src/component/AddItem.js#L54-L56

So it will become

```js
const handleOptionChange = event => {
    setNumberOfDays(+event.target.value);
};
```

If you do that, every time you need to use `numberOfDays` it will be a number already! :)

Reply:
Thank you soo much for going through the code in so much detail!!

I stored all the _magic numbers_ in constants. I also tried converting string to number as you show above (+event.target.value). This works well for adding to the DB, but the only quirk is that after submitting the item, all the radio buttons still become unchecked, and the goal I believe is to always have one checked. So I left the str to num conversion in the addITem function. Since that is giving us the right functionality.

Reply:
That’s totally fine! Thanks for keep improving the codebase! ⭐️